8.x/getting-started/contributing.md

```diff
@@ -385,6 +385,115 @@ ___
+int main(){
+    cout<< "Hello!";
+    return 0;
+}
+```
+
+When you commit these codes, git makes a revision number 0 and assigns it a random string, like 223465:
+
+223465 0 -> Create file main.cpp -> +6 0
+
+`+6 0` means that you added 6 new lines and no line is deleted.
+
+Now you change the code as follows and commit it:
+
+```bash
+#include<iostream>
+int main(){
+    std::cout<< "Hello!";
+    return 0;
+}
+```
+
+Git makes the revision number 1 with a new random string, such as 67fe45:
+
+67fe45 1 -> Do some changes. -> +1 -2
+
+`-2` means that 2 lines are deleted: `using namespace std;` and `cout<< "Hello!";`.
+
+`+1` means that 1 new line is added: `std::cout<< "Hello!";`.
+
+Now you order to revert the last revision:
+
+```bash
+git revert 67fe45
+```
+
+Git does not delete the revision 67fe45, but makes the revision number 2 with a new random string, like 879012:
+
+879012 2 -> Revert the last commit. -> +2 -1
+
+`-1` means that 1 line is deleted: `std::cout<< "Hello!";`.
+
+`+2` means that 2 lines are added: `using namespace std;` and `cout<< "Hello!";`.
+
+So the result is as follows:
+
+```bash
+#include<iostream>
+using namespace std;
+int main(){
+    cout<< "Hello!";
+    return 0;
+}
+```
+
+### Remote
+
+Another interesting aspect of git is that you can put an exact copy of your local repository, with the same master and lateral branches, on another computer in the internet; so others can see and use your codes on their local system. Remote server may be `github` or `gitlab` or any other git based platform. Usually two remote repositories are used: `REMOTE(Origin)` and `REMOTE(Backup)`, so you must select the desired remote repository in order to pull the last changes in your local system.
+
+If you are contributing to the `larammerce-docs` project, you have already forked and cloned the REMOTE(Origin) repository from `https://github.com/larammerce/larammerce-docs`. So the larammerce-docs project on your local system pulls the data from `https://github.com/your-account-name/larammerce-docs` as REMOTE(Origin), while on the teammate's local system pulls from `https://github.com/teammate-account-name/larammerce-docs` as REMOTE(Origin).
+
+For example when you are on `branch_1` in your local system and run `git push --set-upstream origin branch_1`, the branch_1 is pushed from your local system to your REMOTE(Origin) repository, i.e. `https://github.com/your-account-name/larammerce-docs`.
+
+Suppose your teammate creates a new branch, named `sample-branch`, in his local system and pushes it to his REMOTE(Origin) repository and requests to merge it into the master branch of the larammerce-docs project. In order to have your teammate's sample-branch on your local system, create a new remote, for example `REMOTE(Teammate-Origin)`, which pulls the data from `https://github.com/teammate-account-name/larammerce-docs` into your local repository. Run the following command on the master branch of your local repository:
```

Comment:
In order to have your teammate's sample-branch on your local system :
 Consider that it helps us to build that branch on our local repo and see the rendered result.

create a new remote, with a desired name, through which you can pull some desired branches from your teammate's repository into your local repo.